src/main/java/decodes/cwms/algo/ResEvapAlgo.java

```diff
@@ -221,72 +278,78 @@ public double[] getProfiles() throws Exception {
+						CTProfile = new CTimeSeries(newTSID);
+					}
+					else{
+						throw new DbIoException("failed to load time series from database", check.getFailure());
+					}
+
+					TimedVariable newTV = new TimedVariable(new Variable(wtp[i]), CurrentTime);
+					newTV.setFlags(VarFlags.TO_WRITE);
+					CTProfile.addSample(newTV);
+					newWTP.tseries.addTimeSeries(CTProfile);
 				}
-					catch (Exception ex){
+				catch (Exception ex){
 					throw new DbCompException("failed to create new timeSeriesID"+ex);
 				}
 			}
 			else{
-				// todo might need to change
-				CTimeSeries CTProfile = newWTP.getTimeSeriesAt(i);
-				CTProfile.addSample(new TimedVariable(new Variable(wtp[i]), CurrentTime));
+				CTimeSeries CTProfile = newWTP.tseries.getTimeSeriesAt(i);
+				TimedVariable newTV = new TimedVariable(new Variable(wtp[i]), CurrentTime);
+				newTV.setFlags(VarFlags.TO_WRITE);
+				CTProfile.addSample(newTV);
 			}
 			currentDepth += depth_increment;
 		}
 	}
 
+	//Saves last hourly time slice to dailyWTP object
 	public void setDailyProfiles(Date CurrentTime) throws DbCompException {
-		double[] arrayWTP = new double[hourlyWTP.size()];
+		double[] arrayWTP = new double[hourlyWTP.tseries.size()];
 		int i = 0;
-		Date newTime = CurrentTime;
-		for(CTimeSeries CTS: hourlyWTP.getAllTimeSeries()){
+		for(CTimeSeries CTS: hourlyWTP.tseries.getAllTimeSeries()){
 			try{
-				arrayWTP[i] = CTS.findPrev(CurrentTime).getDoubleValue();
-				newTime = CTS.findPrev(CurrentTime).getTime();
+				int idx = CTS.findNextIdx(CurrentTime);
+				if(idx == -1){
+					break;
+				}
+				arrayWTP[i] = CTS.sampleAt(idx).getDoubleValue();
 			}catch(NoConversionException ex){
 				throw new DbCompException("Failed to load value from timeseries"+ex);
 			}
 			i++;
 		}
-		setProfiles(DailyWTP, arrayWTP, newTime);
+		setProfiles(DailyWTP, arrayWTP, CurrentTime, WtpTsid);
 	}
 
-	public void calcDailyEvap(NamedVariable output, CTimeSeries tsc,  Date CurrentTime) throws DbCompException, NoConversionException, DecodesException {
-		double TotalEvap = calcDailyACC(tsc, CurrentTime);
-		Date LastTime = tsc.findPrev(CurrentTime).getTime();
-		setOutput(output, TotalEvap, LastTime);
-		SetAsFlow(TotalEvap, LastTime);
-	}
-
-	public double calcDailyACC(CTimeSeries tsc,  Date CurrentTime) throws DbCompException {
-		TimedVariable loopVar = tsc.findPrev(CurrentTime);
-		Date loopTime = loopVar.getTime();
-		Date pastTime;
-		double total = 0;
-		do{
-			try {
-				total += loopVar.getDoubleValue();
-				pastTime = loopTime;
-				loopVar = tsc.findPrev(loopTime);
-				loopTime = loopVar.getTime();
-
-			}
-			catch(Exception ex){
-				throw new DbCompException("Failed to load past Evap values"+ex);
-			}
-		}while(loopTime.getDate() == pastTime.getDate());
-
-		return total;
-	}
+	//Returns Converted double of cts from currUnits space to NewUnits
 	public double convertUnits(double cts, String currUnits, String newUnits) throws NoConversionException, DecodesException {
```

Comment:
? The time series system contains a unit conversion method. 

NOTE: that's for units of the same abstract measurement (e.g. feet to meters). If this is doing something more than that the name "convertUnits" is insufficient.